bikeshed/unsortedJunk.py

```diff
@@ -1671,3 +1672,64 @@ def forceCrossorigin(doc):
         ):
             continue
         el.set("crossorigin", "")
+
+
+def addImageSize(doc):
+    if doc.md.imgAutoSize is False:
+        return
+    imgElements = findAll("img", doc)
+    for el in imgElements:
+        if el.get("width") or el.get("height") or el.get("noautosize"):
+            continue
+        if el.get("noautosize") is not None:
+            el.attrib.pop("noautosize")
+            continue
+        src = el.get("src")
+        srcset = el.get("srcset")
+        res = 1
+        if src is None and srcset is None:
+            warn("<img> elements must have at least one of src or srcset.")
+            continue
+        elif src is not None and srcset is not None:
+            continue
+        elif src is None:
+            m = re.match(r"^[ \t\n]*([^ \t\n]+)[ \t\n]+(\d+)x[ \t\n]*$", srcset)
+            if m is None:
+                die(
+                    "Couldn't parse srcset attribute: {0}\nBikeshed only supports a single image followed by an integer resolution.",
+                    srcset,
+                )
+                continue
+            else:
+                src = m.group(1)
+                el.set("src", src)
+                res = int(m.group(2))
+        imgPath = parse.urljoin(doc.inputSource.sourceName, src)
```

Comment:
You want to use the inputSource's `relative()` method here, it'll do the correct thing more often. However, it'll still be wrong if your local file references an img URL, so you want a more explicit test for URLs actually - probably just checking if it starts with `http:`, `https:`, or `/`.

Reply:
added the explicit check

Reply:
…as well as switched to using `relative()`, as suggested.